src/sources/http_client/client.rs

```diff
@@ -58,6 +59,13 @@ pub struct HttpClientConfig {
     #[configurable(metadata(docs::human_name = "Scrape Interval"))]
     pub interval: Duration,
 
+    /// The timeout for each scrape request, in seconds.
+    #[serde(default = "default_target_timeout")]
+    #[serde_as(as = "serde_with::DurationSeconds<u64>")]
```

Comment:
It could be nice to allow the user to specify fractional seconds here.

```suggestion
    #[serde_as(as = "serde_with::DurationSecondsWithFrac<f64>")]
```